doc/rst/technotes/libraries.rst

```diff
@@ -551,6 +554,140 @@ include the chapel runtime and standard modules for use in a non-Chapel program
+
+This can be accomplished either by explicitly adding the arguments in the C
+client program itself, or by passing them as arguments to the executable.
+
+The following example programs use the ``foo`` library.  This first example
+demonstrates explicitly adding the arguments in the program:
+
+.. code-block:: C
+
+   #include "foo.h"
+
+   int main(int argc, char* argv[]) {
+     int argChapelC = 3;
+     char* argChapelV[3] = {argv[0], "-nl", "2"};
+     // Initialize the Chapel runtime and standard modules
+     chpl_library_init(argChapelC, argChapelV);
+
+     baz(7); // Call into a library function
+
+     chpl_library_finalize();
+
+     return 0;
+   }
+
+Alternatively, the original single locale client from `Initializing Your
+Library In C`_ can be used with the additional two arguments to the executable:
+
+.. code-block:: bash
+
+   ./a.out -nl 2
+
+Users also still need to call the generated module initialization function for
+multilocale libraries, as mentioned in that section.
+
+In Python
+~~~~~~~~~
+
+Users must still call ``chpl_setup()`` before utilizing the exported Chapel
+functions.  However, it requires a ``numLocales`` argument when the library
+has been compiled for multilocale settings.
+
+Makefile-less Compilation
+-------------------------
+
+When compiling a C program using a multilocale Chapel library without a
+makefile, some additional steps are needed beyond those required for using a
+:ref:`single locale Chapel library <Makefileless Compilation In Single Locale>`.
+
+A new ``compileline`` tool, ``compileline --multilocale-lib-deps`` is currently
+required after ``compileline --libraries``.  Additionally, the library being
+linked must be listed twice in the compilation command - once before
+``compileline --libraries`` and once at the end of the command.  The compilation
+command would then look like this (replacing ``myCProg.c`` with the name of your
+C program that will use the library):
+
+.. code-block:: sh
+
+   `$CHPL_HOME/util/config/compileline --compile` myCProg.c -Llib/ -lfoo `$CHPL_HOME/util/config/compileline --libraries` `$CHPL_HOME/util/config/compileline --multilocale-lib-deps` -lfoo
+
+What is this _server_real program?
+----------------------------------
+
+When you compile a Chapel library for use with multiple locales, you should
+typically see both a library (see :ref:`Location of the Generated Library` for
+where this will be placed and how to control that location) and a binary (which
+is currently generated in the same directory as the "main" source file), in
+addition to other support files such as the generated header, makefile, etc.
+The library will appear like a normal single locale library in terms of the
+interface it provides to client programs - however, under the covers it will
+launch the binary and then communicate with it.  The binary will be what
+executes the exported functions and will communicate the result back to the
+library, to return to the client program.
```

Comment:
I was expecting this paragraph to say how the program is named. Is the program name based upon the library? Or is it always _server_real right now?

Reply:
The program name is based on the library, I'll add that!

Reply:
Okay, I've added some verbage about that